### PR TITLE
View Puzzle screen, wandering creatures and campfires should not be displayed

### PR DESCRIPTION
**Pull request by shprotru**
fix for #5780

@oleg-derevenetz @ihhub @idshibanov, how do you see, guys, the proper way to move this condition:
```
if ( !isPuzzleDraw && mp2_object == MP2::OBJ_CAMPFIRE ) {
            renderMainObject( dst, area, mp );
}
```
inside `isHiddenForPuzzle` function?

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @shprotru , this is wrong way to do as campfire sprite is more than 1 tile in size and not every tile has this object type.

**Comment by ihhub** [comment]
The solution for campfire should be:
```cpp
bool MP2::isHiddenForPuzzle( const int terrainType, uint8_t tileset, uint8_t index )
{
    const int icnID = tileset >> 2;
    if ( icnID < 22 || icnID == 46 || ( icnID == 56 && index == 140 ) || ( icnID == 59 && index >= 14 && index <= 137 ) ) {
        return true;
    }

    return isDiggingHoleSprite( terrainType, tileset, index );
}
```

File: src/fheroes2/gui/interface_gamearea.cpp [hunk 1]
```diff
@@ -464,6 +464,10 @@ void Interface::GameArea::Redraw( fheroes2::Image & dst, int flag, bool isPuzzle
             }
 
             case MP2::OBJ_MONSTER: {
+                if ( !drawHeroes ) {
```
Comment (ihhub) [inline src/fheroes2/gui/interface_gamearea.cpp:467]: Are we certainly sure that this is correct for all cases? What's about View spells and View World?

Comment (shprotru) [reply src/fheroes2/gui/interface_gamearea.cpp:467]: we can add here additional check for `isPuzzleDraw`:

```
if ( isPuzzleDraw && !drawHeroes ) {
                    continue;
}
```

will be this better?

Comment (ihhub) [reply src/fheroes2/gui/interface_gamearea.cpp]: Personally, it feels like a hack which we must avoid. To do it properly we need to add another level of drawings like `LEVEL_MONSTERS` and use it properly without any mentioning of heroes or puzzle.

Comment (ihhub) [reply src/fheroes2/gui/interface_gamearea.cpp]: Let's do an easy way:
```cpp
if ( isPuzzleDraw ) {
    // Monsters should not appear on puzzle map.
    continue;
}
```

**Comment by ihhub** [comment]
@shprotru , thank you very much for the fix!

**Pull request merged**